packages/ember-glimmer/lib/environment.js

```diff
@@ -74,9 +78,17 @@ export default class Environment extends GlimmerEnvironment {
 
     this._definitionCache = new Cache(2000, ({ name, source, owner }) => {
       let { component: componentFactory, layout } = lookupComponent(owner, name, { source });
+      let customManager = undefined;
 
       if (componentFactory || layout) {
-        return new CurlyComponentDefinition(name, componentFactory, layout);
+        if (GLIMMER_CUSTOM_COMPONENT_MANAGER) {
+          let managerId = layout && layout.meta.managerId;
+
+          if (managerId) {
+            customManager = owner.resolveRegistration(`component-manager:${managerId}`);
```

Comment:
I think in general we want either `.lookup` or `.factoryFor().class` instead of `resolveRegistration`.